ckanext/datastore/writer.py

```diff
@@ -140,7 +149,21 @@ def __init__(self, output, columns):
         self.id_col = columns[0] == u'_id'
         if self.id_col:
             columns = columns[1:]
+        # (canada fork only): conform to XML stadards for element names
+        # TODO: upstream contrib??
+        #
+        # Element names must start with a letter or underscore.
+        # Element names cannot start with the letters xml (or XML, or Xml, etc).
+        # Element names can contain letters, digits, hyphens, underscores, and periods.
+        # Element names cannot contain spaces.
+        #
         self.columns = columns
+        self.element_names = {}
+        for col in columns:
+            element_name = col
+            for rule, replacement in xml_element_name_rules:
+                element_name = re.sub(rule, replacement, element_name)
+            self.element_names[col] = element_name
```

Comment:
after this outer for loop you need to look for duplicates and add a counter or something to prevent them